pokedex-with-bit/app_py.py

```diff
@@ -0,0 +1,102 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import shutil
+import click
+import sys
+from glob import glob
+
+from jina import Flow, DocumentArray
+from jina.types.document.generators import from_files
+from jina.logging import default_logger as logger
+from executors import *
+
+MAX_DOCS = int(os.environ.get('JINA_MAX_DOCS', 50000))
+IMAGE_SRC = 'data/**/*.png'
+
+os.environ['JINA_WORKSPACE'] = './workspace'
+os.environ['JINA_PORT'] = os.environ.get('JINA_PORT', str(45678))
+
+
+def index(num_docs: int):
+    num_docs = min(num_docs, len(glob(os.path.join(os.getcwd(), IMAGE_SRC),
+                                      recursive=True)))
+
+    flow = Flow(workspace="workspace")\
+        .add(uses={"jtype": "ImageCrafter",
+                   "with": {"target_size": 96,
+                            "img_mean": [0.485, 0.456, 0.406],
+                            "img_std": [0.229, 0.224, 0.225]}})
+    flow = flow.add(uses=BigTransferEncoder)
+    flow = flow.add(uses={"jtype": "EmbeddingIndexer",
+                          "with": {"index_file_name": "image.json"},
+                          "metas": {"name": "vec_idx"}},
+              name="vec_idx")
+    flow = flow.add(uses={"jtype": "KeyValueIndexer",
+                          "metas": {"name": "kv_idx"}},
+                    name="kv_idx",
+                    needs="gateway")    # to enable parallel running
+    flow = flow.add(name="join_all",
+                    needs=["kv_idx", "vec_idx"],
+                    read_only="true")
```

Comment:
These can be chained together, I'm not sure which way is better.

Reply:
https://github.com/jina-ai/examples/pull/626/commits/473cafec51508030ab3c81c8bfa4cc08439a92ec